network/p2p/streams.go

```diff
@@ -73,7 +73,14 @@ func (n *streamManager) streamHandler(stream network.Stream) {
 				n.log.Infof("Failed to check old stream with %s: %v", remotePeer, err)
 			}
 			n.streams[stream.Conn().RemotePeer()] = stream
-			n.handler(n.ctx, remotePeer, stream, true)
+			if stream.Stat().Direction == network.DirUnknown {
```

Comment:
this `incoming` determination code is questionable, since this function is called with a new stream is accepted by libp2p, it helped to understand a true connection/stream directions when debugging